include/alpaka/math/acos/AcosUniformCudaHipBuiltIn.hpp

```diff
@@ -30,22 +30,16 @@ namespace alpaka
         {
             //! The CUDA acos trait specialization.
             template<typename TArg>
-            struct Acos<AcosUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point<TArg>::value>>
+            struct Acos<AcosUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
             {
                 __device__ auto operator()(AcosUniformCudaHipBuiltIn const& acos_ctx, TArg const& arg)
                 {
                     alpaka::ignore_unused(acos_ctx);
-                    return ::acos(arg);
-                }
-            };
 
-            template<>
-            struct Acos<AcosUniformCudaHipBuiltIn, float>
-            {
-                __device__ auto operator()(AcosUniformCudaHipBuiltIn const& acos_ctx, float const& arg) -> float
-                {
-                    alpaka::ignore_unused(acos_ctx);
-                    return ::acosf(arg);
+                    if constexpr(std::is_same_v<TArg, float>)
+                        return ::acosf(arg);
+                    else
+                        return ::acos(arg);
```

Comment:
Same as above. IMO we should always be explicit and assert for other data types. This would avoid implicit casts e.g. if the user is using `long double`

Reply:
`long double` is treated as `double` in CUDA device code (nvcc warns about this). Otherwise all `is_floating_point` cases should be handled, right?

Reply:
We do not have only one CUDA compiler but the same issue is happening with `half`, ...